src/topic.rs

```diff
@@ -1,21 +1,36 @@
+use smallstr::SmallString;
+
 /// A message queue topic name to which messages can be published
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
-pub struct Topic(&'static str);
+// A survey of common topics found lengths between 16 and 35 bytes
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Topic(SmallString<[u8; 36]>);
```

Comment:
is this saying we can support any string up to 36Bytes? I wonder if any of our topics exceed that? Or is this similar to a stackvec and will falback to heap allocation if we exceed that threshold?

Reply:
It's backed by SmallVec, which yeah does the fallback to heap